src/ConfigProvider.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\View;
+
+use Laminas\ServiceManager\ConfigInterface;
+use Laminas\ServiceManager\Factory\InvokableFactory;
+use Laminas\ServiceManager\ServiceManager;
+
+use function array_merge;
+
+/**
+ * @psalm-import-type FactoriesConfigurationType from ConfigInterface
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
+ * @psalm-type ViewHelperConfigurationType = array{
+ *     view_helpers?: ServiceManagerConfiguration,
+ *     view_helper_config?: array<string, array<array-key, mixed>
+ * }
+ * @psalm-suppress DeprecatedClass
+ */
+final class ConfigProvider
+{
+    /** @return ViewHelperConfigurationType */
+    public function __invoke(): array
+    {
+        return [
+            'view_helpers'       => self::viewHelperDependencyConfiguration(),
+            'view_helper_config' => [],
+        ];
+    }
+
+    /**
+     * @psalm-suppress DeprecatedClass
+     * @return ServiceManagerConfiguration
+     */
+    public static function viewHelperDependencyConfiguration(): array
+    {
+        return [
+            'factories' => self::defaultViewHelperFactories(),
+            'aliases'   => self::defaultViewHelperAliases(),
+        ];
+    }
+
+    /**
+     * Default factories
+     *
+     * basepath, doctype, and url are set up as factories in \Laminas\Mvc\ ViewHelperManagerFactory.
+     * basepath and url are not very useful without their factories, however the doctype
+     * helper works fine as an invokable. The factory for doctype simply checks for the
+     * config value from the merged config.
+     *
+     * @return FactoriesConfigurationType
+     */
+    public static function defaultViewHelperFactories(): array
```

Comment:
Please mark this as `@internal` since there should be no reason to use this outside of this repository.
This also enables us to remove it in the next major.

Therefore, adding `@deprecated` would be helpful so we can find it when preparing the next major.